File: evaluation_for_comparators/baseline_transcrypt/mathgen/tracer_skip.py
import hashlib
import math
seed = 1
def user_hash_random():
    global seed
    h = hashlib.sha256()
    h.update(str(seed).encode('utf-8'))
    seed += 1
    return int(h.hexdigest(), 16) / (2 ** 256 - 1)

def user_randint(min, max):
    return math.floor(user_hash_random() * (max - min + 1)) + min

def user_choice_func1(x):
    if user_hash_random() < 0.5:
        return x[0]
    else:
        return x[1]

def user_choice_func2(x):
    return x[math.floor(user_hash_random() * len(x))]

def user_sample_func1(x, n):
    # This sample function works assuming x is in the form of range(a, b)
    a = x[0]
    b = x[len(x)-1] + 1

    lst = []
    for i in range(n):
        lst.append(math.floor(user_hash_random() * (b - a + 1)) + a)
    return lst

def user_sample_func2(x, n):
    x = list(x)
    lst = []
    for _ in range(n):
        index = math.floor(user_hash_random() * len(x))
        lst.append(x[index])
        del x[index]
    return lst

def user_uniform(a, b):
    return user_hash_random() * (b - a) + a

def user_reset_seed():
    global seed
    seed = 1

def get_seed():
    return seed

def set_seed(new_seed):
    global seed
    seed = new_seed

File: evaluation_for_comparators/baseline_gpt4/mathgen/tracer_skip.js
const crypto = require('crypto');
const { exp } = require('numjs');
var seed = 1;
function user_hash_random(){
  const hash = crypto.createHash('sha256');
  hash.update(seed.toString());
  seed += 1;
  const hex = hash.digest('hex');
  return parseInt(hex, 16) / (2 ** 256 - 1);
}

function user_randint(min, max) {
  return Math.floor(user_hash_random() * (max - min + 1)) + min;
}

function user_choice_func1(x) {
  if (user_hash_random() < 0.5) {
    return x[0];
  } else {
    return x[1];
  }
}

function user_choice_func2(x) {
  return x[Math.floor(user_hash_random() * x.length)];
}

function user_sample_func1(x, n) {
  // This sample function user_works assuming x is in the form of range(a, b)
  var a = x[0];
  var b = x[x.length - 1] + 1;

  var lst = [];
  for (var i = 0; i < n; i++) {
    lst.push(Math.floor(user_hash_random() * (b - a + 1)) + a);
  }
  return lst;
}

function user_sample_func2(x, n) {
    x = Array.from(x);
    var lst = [];
    for (var i = 0; i < n; i++) {
        var index = Math.floor(user_hash_random() * x.length);
        lst.push(x[index]);
        x.splice(index, 1);
    }
    return lst;
}

function user_uniform(a, b) {
    return user_hash_random() * (b - a) + a
}

function user_reset_seed() {
    seed = 1;
}

function get_seed() {
    return seed;
}

function set_seed(new_seed) {
    seed = new_seed;
}

function factorial(a) {
  let result = 1;
  for (let i = 2; i <= a; i++) {
      result *= i;
  }
  return result;
}

function greatest_common_divisor(number1, number2){
  var _return_value;
  number1 = Math.abs(number1);
  number2 = Math.abs(number2);
  while (number2 > 0) {
    var temp = number1;
    number1 = number2;
    number2 = temp % number2;
  }
  _return_value = number1;
  return _return_value;
}

exports.user_hash_random = user_hash_random;
exports.user_randint = user_randint;
exports.user_choice_func1 = user_choice_func1;
exports.user_choice_func2 = user_choice_func2;
exports.user_sample_func1 = user_sample_func1;
exports.user_sample_func2 = user_sample_func2;
exports.user_uniform = user_uniform;
exports.user_reset_seed = user_reset_seed;
exports._factorial = factorial;
exports._greatest_common_divisor = greatest_common_divisor;
exports.get_seed = get_seed;
exports.set_seed = set_seed;
